base/src/ImageEncoderCV.cpp

```diff
@@ -56,15 +56,25 @@ class ImageEncoderCV::Detail
 
 	}
 
+	// void executeCallback() {
+	// 	if (mErrorCallback) 
+	// 	{
+	// 		ErrorObject error(0, "Error: Testing ImageEncoder CV", mModuleName,
+	// 			    		  mModuleId);
+	// 		mErrorCallback(error);
+	// 	}
+	// }
+
 	framemetadata_sp mOutputMetadata;
 	std::string mOutputPinId;
 	cv::Mat iImg;
 	vector<int> flags;
-
+	ErrorCallback mErrorCallback;
+    std::string mModuleId;
```

Comment:
why are these id and name requried here

Reply:
To Propogate this information to Error Callback